src/k8s-engine/__tests__/k3sHelper.spec.ts

```diff
@@ -265,4 +266,129 @@ describe(K3sHelper, () => {
       await pendingInit;
     });
   });
+
+  describe('substituting versions', () => {
+    let subject: K3sHelper;
+    let config: settings.Settings;
+
+    beforeAll(() => {
+      config = settings.defaultSettings;
+      subject = new K3sHelper('x86_64');
+      subject['pendingInitialize'] = Promise.resolve();
+    });
+    const convertToSemvers = function(a: [string, string, string]) : [semver.SemVer, semver.SemVer, semver.SemVer] {
+      return [new semver.SemVer(a[0]), new semver.SemVer(a[1]), new semver.SemVer(a[2])];
+    };
+
+    describe('pickClosestVersion', () => {
+      test('picks same major version before pivot', () => {
+        const semvers = convertToSemvers(['v1.2.3', 'v1.3.4', 'v2.0.0']) ;
+
+        expect(subject['pickClosestVersion'](...semvers)).toBe(semvers[0]);
```

Comment:
It might be clearer to say `expect(…).toHaveProperty('version', 'v1.2.3');`.

Reply:
Done as part of tablizing the tests.